tests/test-upgrade.py

```diff
@@ -0,0 +1,82 @@
+import os
+import time
+from validators import validate_dns, validate_dashboard, validate_storage, validate_ingress
+from subprocess import check_call
+from utils import microk8s_enable, wait_for_pod_state, microk8s_disable, wait_for_installation
+
+upgrade_from = os.environ.get('UPGRADE_MICROK8S_FROM', 'beta')
+# Have UPGRADE_MICROK8S_TO point to a file to upgrade to that file
+upgrade_to = os.environ.get('UPGRADE_MICROK8S_TO', 'edge')
+
+
+class TestUpgrade(object):
+    """
+    Validates a microk8s upgrade path
+    """
+
+    def test_upgrade(self):
+        """
+        Deploy, probe, upgrade, validate nothing broke.
+
+        """
+        cmd = "sudo snap install microk8s --classic --{}".format(upgrade_from).split()
```

Comment:
I find it useful to output information about the upgrade_from and upgrade_to here so you know based on the log what it tried to do. Maybe:
```
Testing upgrade from candidate to edge
```